Sources/MessagingInApp/MessagingInAppImplementation.swift

```diff
@@ -17,33 +17,33 @@ class MessagingInAppImplementation: MessagingInAppInstance {
         self.logger = diGraph.logger
         self.inAppProvider = diGraph.inAppProvider
         self.threadUtil = diGraph.threadUtil
-        self.busEventManager = EventBusHandler(eventBus: diGraph.eventBus, eventStorage: diGraph.eventStorage)
+        self.busEventManager = diGraph.eventBusHandler
         initialize()
     }
 
     private func initialize() {
         inAppProvider.initialize(siteId: moduleConfig.siteId, region: moduleConfig.region, delegate: self)
 
-        busEventManager.onReceive(ProfileIdentifiedEvent.self) { event in
+        busEventManager.addObserver(ProfileIdentifiedEvent.self) { event in
             self.logger.debug("registering profile \(event.identifier) for in-app")
 
             self.inAppProvider.setProfileIdentifier(event.identifier)
-        }.store(in: &subscriptions)
```

Comment:
I believe that after removing this line of code, we can also remove other parts of the code? Such as:

```swift
import Combine

private var subscriptions: Set<AnyCancellable> = []
```

Reply:
Its in later PRs, where there is alot more removals.